app/src/main/java/com/android/voyageur/ui/overview/AddTrip.kt

```diff
@@ -372,3 +466,16 @@ fun DatePickerModal(
         DatePicker(state = datePickerState)
       }
 }
+
```

Comment:
Thank you for implementing the gallery permissions! You did a great job. However, i have a question about the UX functionality in this case:

When user selects "Limited Access" on the pop up dialog:

1. System permission dialog appears
2.  User doesn't select any photos, but clicks on the top left "x"
3. App shows "Please allow access" message instead of handling the cancellation gracefully

 ![WhatsApp Image 2024-11-11 at 11 06 20 AM](https://github.com/user-attachments/assets/b239db7d-be87-4853-bde0-7a20a89f563a)
 ![WhatsApp Image 2024-11-11 at 11 06 20 AM(1)](https://github.com/user-attachments/assets/b562c9df-593f-4b61-bdda-2c839394e5da)
 ![WhatsApp Image 2024-11-11 at 11 06 21 AM(1)](https://github.com/user-attachments/assets/a385cb93-d429-4c13-919f-5e0ce019e182)

Is it the correct behavior? I think it would be cleaner to handle this case differently, for example treating cancellation as a valid user action, without showing any error message. 
Let me know! :)

Reply:
Thanks for the comment. I find this behaviour is more User Friendly (I find that if the user presses on X, they don't choose any photo which should be allowed, thus the Toast asking the user to allow for permission again). I also discussed this with Bogdan, who implemented this for EditProfileScreen, and we agreed this is the correct behaviour.